sstables/mc/writer.cc

```diff
@@ -1382,10 +1383,15 @@ stop_iteration writer::consume_end_of_partition() {
         _first_key = *_partition_key;
     }
     _last_key = std::move(*_partition_key);
+    _partition_key = std::nullopt;
```

Comment:
Can we also (cheaply) validate other invariants, like partition start emitted before clustering rows are, static row is emitted right after partition start, etc? Not proposing doing these in this PR.

Reply:
Yes, will do as a follow up